src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -1291,3 +1334,29 @@ PartitionBound(Oid partitionId)
 
 	return partitionBoundString;
 }
+
+
+/*
+ * ListShardsUnderParentRelation returns a list of ShardInterval for every
+ * shard under a given relation, meaning it includes the shards of child
+ * tables in a partitioning hierarchy. Assumes caller holds all necessary locks.
+ */
+List *
+ListShardsUnderParentRelation(Oid relationId)
+{
+	List *shardList = LoadShardIntervalList(relationId);
+
+	if (PartitionedTableNoLock(relationId))
+	{
+		List *partitionList = PartitionListNoLock(relationId);
```

Comment:
I don't think we need to use NoLock versions here. We're comfortably past having the ShareUpdateExclusive lock.